src/dodal/utils.py

```diff
@@ -76,6 +81,9 @@ def __post_init__(self):
         self.insertion_prefix = f"SR{self.ixx[1:3]}{self.suffix}"
 
 
+T = TypeVar("T", bound=AnyDevice)
```

Comment:
> Maybe I'm misunderstanding here, but this gives off the impression that you could use the skip_device decorator for any functions that return callables, even, e.g.

That's precisely what `bound=` controls - i.e. that `T` can only be any instance of a v1 or v2 device and not `int`?

I think the type annotation is useful as `T` rather than `AnyDevice` because if you have:

```python
@skip_device
def blah() -> SomeV1Device:
    ...
```
then without the extra generic type hinting on `skip_device` we lose the information that we get back `SomeV1Device` and instead they type-checker only knows that `blah()` returns `AnyDevice`.

As a concrete example, with the type extra type hints this is valid:

![image](https://github.com/DiamondLightSource/dodal/assets/24375813/5210b823-60fc-43c8-b8ea-7dfcab3bc2eb)

but with an `AnyDevice` return type on `skip_device` it isn't:

![image](https://github.com/DiamondLightSource/dodal/assets/24375813/5a76fc8f-d2b7-4d79-8952-00b46c9f0cc3)

I'd argue that for "library"-type classes like this one it's good to get the typing right. I care much less about it in "user" type code.

Reply:
Ah, ok I see. Yep; this makes a lot of sense.